checkov/secrets/plugins/custom_regex_detector.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+from typing import Set, Any, Generator, Pattern, Optional, Dict, Tuple, cast, List
+
+import yaml  # type: ignore
+from detect_secrets.constants import VerifiedResult
+from detect_secrets.core.potential_secret import PotentialSecret
+from detect_secrets.plugins.base import RegexBasedDetector
+import re
+
+from checkov.common.bridgecrew.platform_integration import bc_integration
+from detect_secrets.util.code_snippet import CodeSnippet
+from detect_secrets.util.inject import call_function_with_arguments
+
+
+DETECTORS_BY_CUSTOMER_CACHE: dict[str, list[dict[str, Any]]] = {}
```

Comment:
Do we want to handle cache invalidation in some way?
Meaning, add timestamp and invalidate too old entries?